src/backend/distributed/executor/multi_utility.c

```diff
@@ -3819,6 +3819,15 @@ ProcessDropTableStmt(DropStmt *dropTableStatement)
 			continue;
 		}
 
+		/*
+		 * We should make this check only if we have a partitioned table with at
+		 * least one partition. Because in workers, we are already DETACHing
+		 * partitions via command sent from coordinator. If there is a partitioned
+		 * table with partition and if we are not coordinator, this means user tried
+		 * to DROP partitioned table directly from worker.
+		 */
+		EnsureCoordinator();
```

Comment:
This explanation sounds complicated and I don't completely follow. What I do understand is that we don't want users to drop distributed tables from workers. I think you can omit the comment, since EnsureCoordinator() is reasonably descriptive.